frontend_app_gallery/src/container/Gallery.jsx

```diff
@@ -53,55 +58,147 @@ export class Gallery extends React.Component {
         fileName: '',
         folderParentIdList: []
       },
-      imagesPreviews: [],
-      fileCurrentPage: 1,
-      fileSelected: 0,
+      imagePreviewList: [],
+      displayedPictureIndex: 0,
       autoPlay: null,
       fullscreen: false,
       displayPopupDelete: false,
-      imagesPreviewsLoaded: false,
+      imagePreviewListLoaded: false,
       breadcrumbsLoaded: false
     }
 
     // i18n has been init, add resources from frontend
     addAllResourceI18n(i18n, this.state.config.translation, this.state.loggedUser.lang)
     i18n.changeLanguage(this.state.loggedUser.lang)
 
+    props.registerCustomEventHandlerList([
+      { name: CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE, handler: this.handleAllAppChangeLanguage }
+    ])
+
+    props.registerLiveMessageHandlerList([
+      { entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.CREATED, optionalSubType: TLM_ST.FILE, handler: this.handleContentCreatedOrUndeleted },
+      { entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.MODIFIED, optionalSubType: TLM_ST.FILE, handler: this.handleContentModified },
+      { entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.DELETED, optionalSubType: TLM_ST.FILE, handler: this.handleContentDeleted },
+      { entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.UNDELETED, optionalSubType: TLM_ST.FILE, handler: this.handleContentCreatedOrUndeleted }
+    ])
+
     this.lightboxRotation = new LightboxRotation()
+  }
+
+  // TLM Handlers
+
+  // NOTE - RJ - 2020-06-10
+  // a comment about the handling of live messages here:
+  // https://github.com/tracim/tracim/issues/3107#issuecomment-643994410
 
-    document.addEventListener(CUSTOM_EVENT.APP_CUSTOM_EVENT_LISTENER, this.customEventReducer)
+  liveMessageNotRelevant (data, state) {
+    return Number(data.content.workspace_id) !== Number(state.config.appConfig.workspaceId) ||
```

Comment:
Is there a case where the `data.content.workspace_id` will not be a number?

Reply:
Ah ah xD
See: https://github.com/tracim/tracim/pull/3146/files/3cbfb880f7389e5bac621879f56e8685013c276a#r436679917
where I have written exactly the same thing as you to @Skylsmoi.

I think we should leave this as is for now and maybe write a ticket to solve this? (this is a backend bug, right, @Skylsmoi?

Reply:
I'm okay with not using `Number()` for `data.content.workspace_id` if we are sure it's always a number though

Reply:
The state part I understand it might be from an old code, but the data.content we're adding now with TLM, no? In that case we shouldn't be sure of the type?}

Note: I'm ok with leaving the Number, I'm just asking to understand, because I'm not guaranteeing that it's a number in my PRs, I thought that at least for the `data.content` we could be sure.